tests/syft/lib/torch/tensor/tensor_remote_method_api_suite_test.py

```diff
@@ -232,13 +239,18 @@ def test_all_allowlisted_tensor_methods_work_remotely_on_all_types(
         # Step 13: If there are NaN values, set them to 0 (this is normal for division by 0)
 
         try:
+            # TODO: We should detect tensor vs primitive in a more reliable way
             # set all NaN to 0
-            local_result[local_result != local_result] = 0
-            target_result[target_result != target_result] = 0
-
-            # Step 14: Ensure we got the same result locally (using normal pytorch) as we did remotely
-            # using Syft pointers to communicate with remote torch objects
-            assert (local_result == target_result).all()
+            if isprimitive(value=target_result):
+                assert local_result == target_result
+            else:
+                # type(target_result) == torch.Tensor
+                local_result[local_result != local_result] = 0
```

Comment:
Q: Why we need this?
(can't we have only the assert form the end?)

Reply:
We can't access primitives as array index tensor style and they dont have NaNs to convert to 0 so this just fails on Non tensor types. The != = 0 is short hand for converting NaN to 0 because NaN != NaN